doc/pins/pins_hld.md

```diff
@@ -0,0 +1,194 @@
+
+### Packet IO
+P4RT clients can program ACLs to punt packets received in the ingress pipeline. These packets will be trapped and punted to the P4RT application running in the switch and will be forwarded to the client over grpc. 
+
+SONiC relied on basic packet I/O support that uses netdev ports to send and receive packets. P4Runtime applications require additional packet attributes like the target egress port which are not available via netdev. Similarly directed Tx from application level is not currently supported in the SONiC framework. This is required to support scenarios where the remote SDN controller needs to send a packet via one of the front panel ports of the switch.
+
+To support the above requirements, a model similar to sFlow is used. The receive solution creates a netlink type host interface. It programs a user defined trap for packets that are interesting to the controller and maps them to the netlink host interface. A generic ASIC independent model is defined for passing parameters such as “target egress port” to the application container. A detailed description of the receive path is present in the supplementary document on [Packet IO](#broken-todo).
+
+### Repositories
+This design adds the following new repositories:
+*   P4RT Application
+
+It also has modifications in the following repositories:
+*   SWSS
+*   SONiC Build Image
+
+## SAI API
+There are no changes to SAI. 
+
+For programmable hardware targets, PINS introduces an additional SAI header, saip4ext.h, to map user-defined private P4 extensions to vendor SAI implementations. More details on saip4ext.h can be found in the supplemental document [P4 Extensions for SAI](#broken-todo).
+
+## Configuration and management
+
+### CLI/YANG model Enhancements
+There are no CLI changes.
+
+### Config DB Enhancements
+There are no Config DB changes.
+
+## Warmboot and Fastboot Design Impact
+Similar to other orchagents, the P4Orchagent needs to perform actions for successful warm boot and fast boot operation. These changes are being planned for the next phase ( next SONiC release). In this phase, warmboot and fastboot features will not be supported for objects created through the P4RT application.
+
+When the P4RT application is not used, warm boot and fastboot will continue to operate as before. There will be no impact on existing warmboot and fastboot design.
+
+## Restrictions/Limitations
+In this introductory phase, limited functionality and features will be supported through the P4RT application. The details of supported features are provided in the requirements section.
+
+
+## Testing Requirements/Design
+The entire PINS code, including the P4RT Application, P4 orchagent, etc is well unit tested. The coverage goal is at least 90% which is already achieved. Integration tests will be added in the following release.
```

Comment:
do we have sonic-mgmt intergation test planned?

Reply:
Yes, in the second release.